cmd/snap/cmd_list_test.go

```diff
@@ -116,8 +121,10 @@ func (s *SnapSuite) TestListWithNotes(c *check.C) {
 	rest, err := snap.Parser().ParseArgs([]string{"list"})
 	c.Assert(err, check.IsNil)
 	c.Assert(rest, check.DeepEquals, []string{})
-	c.Check(s.Stdout(), check.Matches, `Name +Version +Rev +Developer +Notes
-foo +4.2 +17 +bar +try
-`)
+	c.Check(s.Stdout(), check.Matches, `(?ms)^Name +Version +Rev +Developer +Notes$`)
+	c.Check(s.Stdout(), check.Matches, `(?ms).*^foo +4.2 +17 +bar +try$`)
+	c.Check(s.Stdout(), check.Matches, `(?ms).*^dm1 +.* +devmode$`)
+	c.Check(s.Stdout(), check.Matches, `(?ms).*^dm2 +.* +devmode$`)
+	c.Check(s.Stdout(), check.Matches, `(?ms).*^cf1 +.* +confined$`)
```

Comment:
I'd expect simple lack of devmode here to mean confined. The presence of confined makes it feel like whatever doesn't say so would be not-confined.